### PR TITLE
Add dialog to navigate long room topics

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes vector-im/element-web#9623

If your power level allows you to edit the room topic, a "Edit Topic" button will appear below the text. Clicking on it will dismiss this modal and show the room settings at the correct tab for one to update the topic

https://user-images.githubusercontent.com/769871/167126443-d8c6e944-0825-419c-a690-d8d84bd3587f.mov

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add dialog to navigate long room topics ([\#8517](https://github.com/matrix-org/matrix-react-sdk/pull/8517)). Fixes vector-im/element-web#9623.

### CODE REVIEW

File: src/hooks/room/useTopic.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
```
Comment (kerryarchibald) [inline src/hooks/room/useTopic.ts:2]: wrong year?

File: src/components/views/elements/RoomTopic.tsx [hunk 2]
```diff
@@ -14,35 +14,91 @@ See the License for the specific language governing permissions and
...
+        if (target.tagName.toUpperCase() === "A") {
+            return;
+        }
+
+        dis.fire(Action.ShowRoomTopic);
+    }, [props]);
+
+    useDispatcher(dis, (payload) => {
+        if (payload.action === Action.ShowRoomTopic) {
+            const canSetTopic = room.currentState.maySendStateEvent('m.room.topic', client.getUserId());
```
Comment (kerryarchibald) [inline src/components/views/elements/RoomTopic.tsx:61]: Could use EventType enum here

**Review by kerryarchibald — APPROVED** [review]
LGTM! Left a couple of small nitpick comments

**Review by niquewoodhouse — COMMENTED** [review]
- [x] Links don't seem to always work in the topic dialog?

It might be helpful if links worked in the dialog, but they don't seem to reliably appear.

https://user-images.githubusercontent.com/1587166/167646801-b7eb22ce-d956-409b-bf23-2a33f356bb2e.mov

**Comment by gsouquet** [comment]
@niquewoodhouse It's odd I don't think I've encountered that when I first developed the solution but could reproduce reliably the problem now.
I've changed the approach and it's now working all fine, everytime. 

Merging now

**Pull request merged**